### PR TITLE
Use lib: es5 to prevent IE 11 breaks

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

Currently we rely on code reviews to prevent ES6 methods and objects, which aren't supported in IE 11, from slipping into our code. Switching all projects which run in the browser to use `"lib": ["es5"]` (and possibly other relevant libs) in tsconfig.json will prevent these issues.

Related changes:
* Replace instances of ES6 things with ES5-compatible things
* Add `"skipLibCheck": true` to tsconfig.json for projects which have dependencies whose typings use ES6 types
* In a couple instances where the usage is safe, add file-level declarations for Set/Map/WeakMap--either for tests (which run in Node not the browser) or one place where it checks for the class's existence before using it

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8414)

### CODE REVIEW

File: packages/dashboard/tsconfig.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "noImplicitAny": true,
     "moduleResolution": "node",
     "preserveConstEnums": true,
-    "lib": ["es2017", "dom"],
+    "skipLibCheck": true,
+    "lib": ["es5", "es2015.promise", "dom"],
```
Comment (ecraig12345) [inline packages/dashboard/tsconfig.json]: @Raghurk I noticed there are several components in dashboard which use Promise. Do you require all consumers to provide a Promise polyfill (for IE 11 support)?

File: packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx [hunk 2]
```diff
@@ -35,15 +35,15 @@ export class ScrollablePaneBase extends BaseComponent<IScrollablePaneProps, IScr
   private _stickyAboveRef = React.createRef<HTMLDivElement>();
   private _stickyBelowRef = React.createRef<HTMLDivElement>();
   private _contentContainer = React.createRef<HTMLDivElement>();
-  private _subscribers: Set<Function>;
-  private _stickies: Set<Sticky>;
+  private _subscribers: Function[];
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx]: Switched Set to arrays here--more verbose but should hopefully have the same behavior

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx [hunk 3]
```diff
@@ -97,11 +97,11 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
     this._onRootRef = this._onRootRef.bind(this);
     this._isEventOnHeader = this._isEventOnHeader.bind(this);
     this._onDropIndexInfo = {
-      sourceIndex: Number.MIN_SAFE_INTEGER,
-      targetIndex: Number.MIN_SAFE_INTEGER
+      sourceIndex: -1,
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: I read through the code and it seems that -1 will achieve the same thing as MIN_SAFE_INTEGER. @laxmikankanala, looks like you added this code in #4857 -- can you confirm that -1 is safe? (or @aditima  or @yiminwu who reviewed)

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Code seems fine to me. As long as column reordering works, particularly at the edges of the table then I think we're OK.

File: packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx (hunk 2 above)
Comment (kenotron) [reply packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx]: Awesome catch!!!

**Comment by kenotron** [comment]
needs rush change :(

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx (hunk 3 above)
Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: @KevinTCoughlin Would you mind helping test that? (Guessing you're more familiar with the intended behavior.)

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Works well in my testing

Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Thanks for checking!

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx [hunk 4]
```diff
@@ -347,7 +347,7 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
   }
 
   private _isValidCurrentDropHintIndex() {
-    return this._currentDropHintIndex! >= 0;
+    return this._currentDropHintIndex >= 0;
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: :+1:

**Comment by ecraig12345** [comment]
Planning to merge this EOD today. If anyone who hasn't reviewed wants to take a look before I merge, please do so now (or leave a comment asking me to wait).

**Comment by dzearing** [comment]
@ecraig12345 Nice work!!!

**Comment by ecraig12345** [comment]
Waiting to merge until I can implement David's suggestion about declaring IE11-compat versions of Set and Map somewhere. (see https://github.com/OfficeDev/office-ui-fabric-react/pull/8421#discussion_r267982373 )

**Comment by ecraig12345** [comment]
Updated the PR to add a `typings` folder at the root with a `custom-global` package inside (with partial Set/Map/WeakMap types), then referenced that extra typing in oufr, experiments, and utilities (can be added elsewhere as needed).

So this time I really *do* intend to merge EOD unless someone objects.

**Comment by ecraig12345** [comment]
Merged. If anyone prefers a different approach to the Set/Map typings, let me know and we can revisit the issue.

**Pull request merged**